source/extensions/filters/common/expr/context.h

```diff
@@ -116,9 +117,16 @@ template <class T> class HeadersWrapper : public google::api::expr::runtime::Cel
       return {};
     }
     auto str = std::string(key.StringOrDie().value());
-    if (!::Envoy::Http::validHeaderString(str)) {
-      // Reject key if it is an invalid header string
-      return {};
+    if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.consistent_header_validation")) {
+      if (!Http::HeaderUtility::headerValueIsValid(str)) {
```

Comment:
The intention here is checking the **header key/name** . So I think we should use utility `headerNameIsValid` instead?

Reply:
yeah good catch - I'd been going off the bug which I think referenced the wrong helper.  looks like same thing with the formatter.

Reply:
Yeah,  after tracking the code into [here](https://github.com/envoyproxy/envoy/blob/e69057a11361359e55f9cd4ba96bf70be1ea5e76/source/common/formatter/http_specific_formatter.cc#L85-L88) , we can see formatter is also using header key/name